pkgs/development/haskell-modules/make-package-set.nix

```diff
@@ -221,26 +221,34 @@ in package-set { inherit pkgs stdenv callPackage; } self // {
     #   , overrides : Defaulted (HaskellPackageOverrideSet)
     #   , modifier : Defaulted
     #   , returnShellEnv : Defaulted
+    #   , hoogle : Defaulted
     #   } -> NixShellAwareDerivation
     # Given a path to a haskell package directory, an optional package name
     # which defaults to the base name of the path, an optional set of source
     # overrides as appropriate for the 'packageSourceOverrides' function, an
     # optional set of arbitrary overrides, and an optional haskell package
     # modifier, return a derivation appropriate for nix-build or nix-shell to
-    # build that package.
+    # build that package. By default a hoogle database is generated if a shell
+    # environment is requested.
     developPackage =
       { root
       , name ? builtins.baseNameOf root
       , source-overrides ? {}
       , overrides ? self: super: {}
       , modifier ? drv: drv
-      , returnShellEnv ? pkgs.lib.inNixShell }:
-      let drv =
-        (extensible-self.extend
-           (pkgs.lib.composeExtensions
-              (self.packageSourceOverrides source-overrides)
-              overrides))
-        .callCabal2nix name root {};
+      , returnShellEnv ? pkgs.lib.inNixShell
+      , withHoogle ? returnShellEnv }:
+      let
+        hoogleOverrides = self: super:
+          pkgs.lib.optionalAttrs withHoogle {
+            ghc = super.ghc // { withPackages = super.ghc.withHoogle; };
+            ghcWithPackages = self.ghc.withPackages;
```

Comment:
Why is this line needed?

Also, I don't quite understand the `ghc = super.ghc // { withPackages = super.ghc.withHoogle; };` line.

What is the effect of doing these overrides?  Could this cause any problems in any way?

Reply:
I actually wrote this years ago and I'm a little sure that it was necessary then. Nevertheless I discovered envFunc today and this is gone :)